src/routes_admin.py

```diff
@@ -29,7 +29,15 @@ def get_course_roster_page(netid, cid):
         @auth.require_admin_status
         def get_course_staff_roster(netid, cid):
             course = db.get_course(cid)
-            return jsonify(admin_ids=course['admin_ids'], staff_ids=course['staff_ids'], user=netid)
+            
+            staff = course['staff']
+            # get all admin_ids by filtering out all non-admins
+            admin = dict(filter(lambda x : x[1] == {'is_admin': True}, staff.items()))
```

Comment:
Similar to a point above, instead of asserting `x[1] == {'is_admin': True}`, change it to `x[1]["is_admin"]`. If we want to be even more careful, we can use `x[1].get("is_admin", False)`.